### PR TITLE
Add rekeyEdek function, deprecate rekeyDocument function

### PR DESCRIPTION
**Pull request by giarc3**
Changes re-keying to be more intuitive - because they document fields aren't needed for re-keying, they shouldn't need to be passed in. This adds a new rekeyEdek function and deprecates the old function.

Re-key example won't work until we release the new version of the TSC.

### CODE REVIEW

File: src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java [hunk 1]
```diff
@@ -463,22 +462,41 @@ public void close() throws IOException {
...
+    @Deprecated
+    public CompletableFuture<EncryptedDocument> rekeyDocument(EncryptedDocument encryptedDocument,
+            DocumentMetadata metadata, String newTenantId) {
+        return this.encryptionService.rekey(encryptedDocument.getEdek(), metadata, newTenantId)
+                .thenApply(newKey -> new EncryptedDocument(encryptedDocument.getEncryptedFields(),
+                        newKey.getEdek()));
+    }
+
+    /**
+     * Re-key a document's encrypted document key (EDEK) to a new tenant. Decrypts the EDEK then re-encrypts it to the new tenant.
```
Comment (coltfred) [inline src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java:488]: document key should be document encryption key

Comment (giarc3) [reply src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java:488]: As mentioned in other PR, this is the wording we use across all TSCs, so not going to change them all for now

**Review by skeet70 — APPROVED** [review]

File: src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java (hunk 1 above)
Comment (BobWall23) [inline src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java:488]:
```suggestion
     * Re-key a document's encrypted document key (EDEK) using a new KMS config. Decrypts the EDEK then re-encrypts it using the specified tenant's current primary KMS config.
```

File: src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java [hunk 2]
```diff
@@ -463,22 +462,41 @@ public void close() throws IOException {
...
+            DocumentMetadata metadata, String newTenantId) {
+        return this.encryptionService.rekey(encryptedDocument.getEdek(), metadata, newTenantId)
+                .thenApply(newKey -> new EncryptedDocument(encryptedDocument.getEncryptedFields(),
+                        newKey.getEdek()));
+    }
+
+    /**
+     * Re-key a document's encrypted document key (EDEK) to a new tenant. Decrypts the EDEK then re-encrypts it to the new tenant. 
+     * The DEK is then discarded. The old tenant and new tenant can be the same in order to re-key the document to the tenant's 
+     * latest primary config.
```
Comment (BobWall23) [inline src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java:490]:
```suggestion

```

File: src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java [hunk 3]
```diff
@@ -463,22 +462,41 @@ public void close() throws IOException {
...
+    public CompletableFuture<EncryptedDocument> rekeyDocument(EncryptedDocument encryptedDocument,
+            DocumentMetadata metadata, String newTenantId) {
+        return this.encryptionService.rekey(encryptedDocument.getEdek(), metadata, newTenantId)
+                .thenApply(newKey -> new EncryptedDocument(encryptedDocument.getEncryptedFields(),
+                        newKey.getEdek()));
+    }
+
+    /**
+     * Re-key a document's encrypted document key (EDEK) to a new tenant. Decrypts the EDEK then re-encrypts it to the new tenant. 
+     * The DEK is then discarded. The old tenant and new tenant can be the same in order to re-key the document to the tenant's
```
Comment (BobWall23) [inline src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java:489]:
```suggestion
     * The DEK is then discarded.
```

**Review by BobWall23 — APPROVED** [review]
Suggested changes to documentation wording - I think it would be good to make it more apparent that the likely use of rekey is to use the new KMS configuration for a tenant.

**Pull request merged**